cypress/e2e/permit2.test.ts

```diff
@@ -67,140 +64,147 @@ describe('Permit2', () => {
-          cy.get(APPROVE_BUTTON).click()
-          cy.get(APPROVE_BUTTON)
-            .should('have.text', `Approve use of ${INPUT_TOKEN.symbol}`)
+          tokenApprovalStub.restore() // allow token approval
+          cy.get(getTestSelector('confirm-swap-button')).click()
+          cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Approve permit')
+          // There should be a successful Approved notification.
+          cy.contains('Approved').should('exist')
+          // permitApprovalStub should reject here, and the modal should revert to the review state.
+          cy.contains('Review Swap')
+            .should('be.visible')
             .then(() => {
               permitApprovalStub.restore() // allow permit approval
+              cy.get(getTestSelector('confirm-swap-button')).click()
 
               // The swap should now be able to proceed, as the permit2 signature will be accepted by the user.
-              cy.get(APPROVE_BUTTON)
-                .click()
-                .then(() => {
-                  const approvalTime = Date.now()
-                  cy.get(APPROVE_BUTTON).should('have.text', 'Approval pending')
-
-                  // There should be a successful Approved notification.
-                  cy.contains('Approved').should('exist')
+              const approvalTime = Date.now()
 
-                  swaps()
+              cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Confirm Swap')
+              cy.contains('Swapped').should('exist')
 
-                  expectTokenAllowanceForPermit2ToBeMax()
-                  expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime)
-                })
+              expectTokenAllowanceForPermit2ToBeMax()
+              expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime)
             })
         })
     })
   })
 
   it('swaps with existing token approval and missing permit approval', () => {
-    cy.hardhat()
-      .then(({ approval, wallet }) => approval.setTokenAllowanceForPermit2({ owner: wallet, token: INPUT_TOKEN }))
-      .then(() => {
-        cy.get(APPROVE_BUTTON)
-          .click()
-          .then(() => {
-            const approvalTime = Date.now()
-
-            swaps()
-
-            expectPermit2AllowanceForUniversalRouterToBeMax(approvalTime)
-          })
-      })
+    cy.hardhat().then(({ approval, wallet, provider }) => {
+      approval.setTokenAllowanceForPermit2({ owner: wallet, token: INPUT_TOKEN })
+      cy.spy(provider, 'send').as('permitApprovalSpy')
+      cy.get('#swap-button').should('not.have.attr', 'disabled')
+      cy.get('#swap-button').click()
+      cy.get(getTestSelector('confirm-swap-button')).click()
+      const approvalTime = Date.now()
+      // This step should appear, but cypress may sign the request too quickly.
+      // cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Approve DAI')
```

Comment:
Is this issue only happening in CI? When I uncomment L139 locally it passes. If it is only happening in CI, have you tried the code below:

```suggestion
      cy.get(getTestSelector('confirm-swap-button')).click().then(() => {
        cy.get(getTestSelector('PendingModalContent-title')).should('have.text', 'Approve DAI')
      })
      const approvalTime = Date.now()
```